src/modules/reporting/components/reporting-report/reporting-report.jsx

```diff
@@ -0,0 +1,310 @@
+/* eslint jsx-a11y/label-has-for: 0 */
+
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import classNames from 'classnames'
+import { Helmet } from 'react-helmet'
+
+import MarketPreview from 'modules/market/components/market-preview/market-preview'
+
+import { BINARY, CATEGORICAL, SCALAR } from 'modules/markets/constants/market-types'
+
+import FormStyles from 'modules/common/less/form'
+import ConfirmStyles from 'modules/common/less/confirm-table'
+import Styles from 'modules/reporting/components/reporting-report/reporting-report.styles'
+
+export default class ReportingReport extends Component {
+
+  static propTypes = {
+    market: PropTypes.object.isRequired,
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      currentStep: 0,
+      isFormValid: true,
+      isMarketValid: null,
+      selectedOutcome: '',
+      stake: '',
+      validations: {
+        isMarketValid: false,
+        stake: false,
+      }
+    }
+
+    this.prevPage = this.prevPage.bind(this)
+    this.nextPage = this.nextPage.bind(this)
+    this.validateIsMarketValid = this.validateIsMarketValid.bind(this)
+    this.validateOutcome = this.validateOutcome.bind(this)
+    this.validateNumber = this.validateNumber.bind(this)
+    this.validateStake = this.validateStake.bind(this)
+  }
+
+  prevPage() {
+    const newStep = this.state.currentStep <= 0 ? 0 : this.state.currentStep - 1
```

Comment:
You could collapse this into the setState argument as well: 

this.setState({ currentStep: this.state.currentStep <= 0 ? 0 : this.state.currentStep - 1 })

Reply:
^ This is just a code style comment — not required, just a terser way to express the same thing.